src/ui/ui_settings/__tests__/ui_settings_mixin_integration.js

```diff
@@ -0,0 +1,194 @@
+import sinon from 'sinon';
+import expect from 'expect.js';
+import Chance from 'chance';
+
+import ServerStatus from '../../../server/status/server_status';
+import Config from '../../../server/config/config';
+
+import * as uiSettingsServiceFactoryNS from '../ui_settings_service_factory';
+import { uiSettingsServiceFactory } from '../ui_settings_service_factory';
+import * as getUiSettingsServiceForRequestNS from '../ui_settings_service_for_request';
+import { getUiSettingsServiceForRequest } from '../ui_settings_service_for_request';
+import { uiSettingsMixin } from '../ui_settings_mixin';
+
+const chance = new Chance();
+
+describe('uiSettingsMixin()', () => {
+  const sandbox = sinon.sandbox.create();
+
+  function setup(options = {}) {
+    const {
+      enabled = true
+    } = options;
+
+    const config = Config.withDefaultSchema({
+      uiSettings: { enabled }
+    });
+
+    // maps of decorations passed to `server.decorate()`
+    const decorations = {
+      server: {},
+      request: {}
+    };
+
+    // mock hapi server
+    const server = {
+      log: sinon.stub(),
+      config: () => config,
+      decorate: sinon.spy((type, name, value) => {
+        decorations[type][name] = value;
+      }),
+    };
+
+    // "promise" returned from kbnServer.ready()
+    const readyPromise = {
+      then: sinon.stub(),
+    };
+
+    const kbnServer = {
+      server,
+      config,
+      status: new ServerStatus(server),
+      ready: sinon.stub().returns(readyPromise),
+    };
+
+    uiSettingsMixin(kbnServer, server, config);
+
+    return {
+      kbnServer,
+      server,
+      decorations,
+      readyPromise,
+      status: kbnServer.status.get('ui settings'),
+    };
+  }
+
+  afterEach(() => sandbox.restore());
+
+  describe('status', () => {
+    it('creates a "ui settings" status', () => {
+      const { status } = setup();
+      expect(status).to.have.property('state', 'uninitialized');
+    });
+
+    describe('disabled', () => {
+      it('disables if uiSettings.enabled config is false', () => {
+        const { status } = setup({ enabled: false });
+        expect(status).to.have.property('state', 'disabled');
+      });
+
+      it('does not register a handler for kbnServer.ready()', () => {
+        const { readyPromise } = setup({ enabled: false });
+        sinon.assert.notCalled(readyPromise.then);
+      });
+    });
+
+    describe('enabled', () => {
+      it('registers a handler for kbnServer.ready()', () => {
+        const { readyPromise } = setup();
+        sinon.assert.calledOnce(readyPromise.then);
```

Comment:
__optional nit:__ maybe it will be safer to add `sinon.assert.calledWithExactly(readyPromise.then, sinon.match.func);` too, to make sure that code adds `resolve` callback.